tests/namespaced.rs

```diff
@@ -1389,6 +1393,86 @@ mod namespaced {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_ztunnel_self_scrape() -> Result<(), anyhow::Error> {
+        let mut manager = setup_netns_test!(Shared);
+
+        // Deploy ztunnel with workload info
+        let zt = manager.deploy_ztunnel(DEFAULT_NODE).await?;
+        let zt_ip = zt.proxy_addresses.inbound.ip();
+
+        // Deploy a client pod in the same node.
+        let client = manager
+            .workload_builder("client", DEFAULT_NODE)
+            .register()
+            .await?;
+
+        // Self-Scrape Check
+        let target = format!("{zt_ip}:15020");
+        info!(%target, "Attempting HBONE self-scrape");
+        run_hbone_metrics_scrape_client(client, manager.resolver(), &target).await?;
+
+        info!("HBONE self-scrape test passed");
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_ztunnel_metrics_scrape_via_hbone() -> Result<(), anyhow::Error> {
+        let mut manager = setup_netns_test!(Shared);
+
+        // Deploy ztunnel with workload info
+        let zt = manager.deploy_ztunnel(DEFAULT_NODE).await?;
+        
+        // Deploy a prometheus client pod in the same node
+        let prometheus = manager
+            .workload_builder("prometheus", DEFAULT_NODE)
+            .identity(identity::Identity::Spiffe {
+                trust_domain: "cluster.local".into(),
+                namespace: "istio-system".into(),
+                service_account: "prometheus".into(),
+            })
+            .register()
+            .await?;
+
+        let zt_ip = zt.proxy_addresses.inbound.ip();
+        let target = format!("{zt_ip}:15020");
+        info!(%target, "Attempting HBONE metrics scrape");
+        run_hbone_metrics_scrape_client(prometheus, manager.resolver(), &target).await?;
+
+        // BYTES_RECV = bytes sent by client = size of GET request (observed as 41 with hyper client)
+        // BYTES_SENT = bytes received by client = size of metrics response (variable, so I removed assertion)
+        const EXPECTED_METRICS_REQ_SIZE: u64 = 41; // Update expected size based on observation
+        let metrics = [
+            (CONNECTIONS_OPENED, 1),
+            (CONNECTIONS_CLOSED, 1),
+            (BYTES_RECV, EXPECTED_METRICS_REQ_SIZE), 
+            // (BYTES_SENT, ???), 
```

Comment:
Seems ok to leave it out for test reliability. The client does validation of the contents. It can do more if we deem the existing state insufficient.